### PR TITLE
Add cron to trigger sync for old seen android channels

### PR DESCRIPTION
**Pull request by norkans7**
Addresses #279

### CODE REVIEW

File: core/msgio/android_test.go [hunk 1]
```diff
@@ -15,53 +11,11 @@ import (
 	"github.com/stretchr/testify/require"
 )
 
-type MockFCMService struct {
```
Comment (norkans7) [inline core/msgio/android_test.go:18]: moved to testsuite to be able to reuse that in other packages

File: core/tasks/channels/cron.go [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+			syncedCount += 1
+		}
+
+	}
+
+	return map[string]any{"synced": syncedCount, "errored": erroredCount}, nil
+
+}
+
+func getOldSeenAndroidChannels(ctx context.Context, db models.DBorTx) ([]models.Channel, error) {
```
Comment (norkans7) [inline core/tasks/channels/cron.go:62]: we could easily move this to models but currently that only needed for this cron only

File: core/tasks/channels/cron.go [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+
+	"github.com/nyaruka/gocommon/dates"
+	"github.com/nyaruka/mailroom/core/models"
+	"github.com/nyaruka/mailroom/core/msgio"
+	"github.com/nyaruka/mailroom/core/tasks"
+	"github.com/nyaruka/mailroom/runtime"
+)
+
+func init() {
+	tasks.RegisterCron("sync_old_seen_android_channels", &SyncOldSeenAndroidCron{})
```
Comment (rowanseymour) [inline core/tasks/channels/cron.go:17]: `sync_android_channels`

File: core/tasks/channels/cron.go [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+}
+
+type SyncOldSeenAndroidCron struct {
```
Comment (rowanseymour) [inline core/tasks/channels/cron.go:20]: `SyncAndroidChannelsCron`

File: core/tasks/channels/cron.go [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+	FCMClient msgio.FCMClient
+}
+
+func (s *SyncOldSeenAndroidCron) AllInstances() bool {
+	return true
+}
+
+func (s *SyncOldSeenAndroidCron) Next(last time.Time) time.Time {
+	return tasks.CronNext(last, time.Minute*5)
```
Comment (rowanseymour) [inline core/tasks/channels/cron.go:29]: RP has this at 10 minutes currently.. any reason to change?

File: core/tasks/channels/cron_test.go [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+	rc := rt.RP.Get()
+	defer rc.Close()
+
+	defer testsuite.Reset(testsuite.ResetAll)
+
+	mockFCM := testsuite.NewMockFCMService("FCMID3", "FCMID4", "FCMID5")
+	fc := mockFCM.GetClient(ctx)
+
+	// make all channel inactive
+	rt.DB.MustExec(`UPDATE channels_channel SET is_active = False WHERE id > 0`)
```
Comment (rowanseymour) [inline core/tasks/channels/cron_test.go:26]: can you rework this test to not modify the std test data? Then you can use `testsuite.Reset(testsuite.ResetData)` above which is a lot faster

**Review by rowanseymour — CHANGES_REQUESTED** [review]

File: core/tasks/channels/cron.go (hunk 5 above)
Comment (norkans7) [reply core/tasks/channels/cron.go:29]: oops mistake I read that wrong

File: core/tasks/channels/cron_test.go (hunk 6 above)
Comment (norkans7) [reply core/tasks/channels/cron_test.go:26]: I am not following on this

Comment (norkans7) [reply core/tasks/channels/cron_test.go:26]: never mind I found that

File: core/tasks/channels/cron.go (hunk 2 above)
Comment (rowanseymour) [reply core/tasks/channels/cron.go:62]: We should move it to models.. it's not just a question of re-use but grouping code that does similar stuff. If we make changes to the db schema of channel for example - it's easier if the functions that query channels from the database are in the same place and their sql are in the same file.

File: core/tasks/channels/cron.go [hunk 7]
```diff
@@ -0,0 +1,94 @@
...
+	}
+
+	return map[string]any{"synced": syncedCount, "errored": erroredCount}, nil
+
+}
+
+func getOldSeenAndroidChannels(ctx context.Context, db models.DBorTx) ([]models.Channel, error) {
+	now := dates.Now()
+	start := now.AddDate(0, 0, -7)
+	end := now.Add(-time.Minute * 15)
```
Comment (rowanseymour) [inline core/tasks/channels/cron.go:65]: We don't need this

**Review by rowanseymour — APPROVED** [review]

**Pull request merged**